packages/expo-eas-client-id/android/src/main/java/expo/modules/easclientid/EASClientID.kt

```diff
@@ -0,0 +1,21 @@
+package expo.modules.easclientid
+
+import android.content.Context
+import java.util.UUID
+
+private const val PREFERENCES_FILE_NAME = "dev.expo.EASSharedPreferences"
+private const val EAS_CLIENT_ID_SHARED_PREFERENCES_KEY = "eas-client-id"
+
+class EASClientID(private val context: Context) {
+  val uuid: UUID by lazy {
+    val sharedPreferences = context.getSharedPreferences(PREFERENCES_FILE_NAME, Context.MODE_PRIVATE)
+    if (!sharedPreferences.contains(EAS_CLIENT_ID_SHARED_PREFERENCES_KEY)) {
+      with(sharedPreferences.edit()) {
+        putString(EAS_CLIENT_ID_SHARED_PREFERENCES_KEY, UUID.randomUUID().toString())
+        apply()
+      }
+    }
+
+    UUID.fromString(sharedPreferences.getString(EAS_CLIENT_ID_SHARED_PREFERENCES_KEY, null) ?: throw Exception("No EAS client ID set"))
```

Comment:
I think the exception is not necessary.

```suggestion
    var id = sharedPreferences.getString(EAS_CLIENT_ID_SHARED_PREFERENCES_KEY, null)
    if (id == null) {
      id = UUID.randomUUID().toString()
      with(sharedPreferences.edit()) {
        putString(EAS_CLIENT_ID_SHARED_PREFERENCES_KEY, id)
        apply()
      }
    }

    UUID.fromString(id)
 ```